odoo/orm/table_objects.py

```diff
@@ -79,7 +83,7 @@ class Constraint(TableObject):
 
     def __init__(
         self,
-        definition: str,
+        definition: DefinitionType,
```

Comment:
Actually, you don't need this for constraints. It would mean you can have different constrains depending on your configuration, which means you cannot trust any of them.

I can stay, but I find it can be confusing.

Reply:
You're right. It's better to avoid having a callable for database constraints.